data-entry/dataEntry.app.js

```diff
@@ -78,12 +78,62 @@
+                                        if (annotation.content.term) {
+                                            termColumnName = annotation.content.term;
+                                        }
+                                    });
+                                    // if a term annotation is defined
+                                    if (termColumnName) {
+                                        // do nothing and break out of "if"
+                                    }
+                                    // use the term column instead
+                                    else if(table.columns._columns.Term) {
+                                        termColumnName = "Term";
+                                    }
+                                    // next choice is name
+                                    else if(table.columns._columns.Name) {
+                                        termColumnName = "Name";
+                                    }
+                                    // find a column with unambigous text field ?
+                                    else {
+                                        // TODO
+                                        // angular.forEach(table.columns.all(), function(column) {
+                                        //
+                                        // });
+                                    };
+
+                                    var foreignKeyTablePath = new ERMrest.DataPath(table);
+                                    foreignKeyTablePath.entity.get().then(function success(entitySet) {
+                                        angular.forEach(values.data, function(value) {
+                                            var field = null;
+                                            if (value[termColumnName]) {
+                                                field = value;
+                                            } else {
+                                                var foreignKey = Object.keys(value)[0];
+                                                // this returns a set
+                                                var filteredSet = $filter('filter')(entitySet,
+                                                    // checks that the foreign key and key are the same
+                                                    function (_value, index, array) {
+                                                        return _value[foreignKey] == value[foreignKey];
+                                                    }
+                                                );
+                                                // there should only be one
+                                                field = filteredSet[0];
+                                            }
+                                            // push the annotation defined field from foreign key table
+                                            domainValues.push(field[termColumnName]);
```

Comment:
You'll need to push at least two columns, the domain values for the foreign key assignment, and the corresponding term/name/etc for display.